### ISSUE TITLE
gsap 1.20.5 server side rendering error

### ISSUE THREAD

**Issue by hadimostafapour** [greensock/GreenSock-JS#268]
after updating to version 1.20.5, I got this error on SSR.

```
webpack:///./node_modules/gsap/TweenLite.js?:42
                        _doc = window.document,
                                      ^

TypeError: Cannot read property 'document' of undefined
```

**Comment by jackdoyle** [comment]
Hm, would you mind sending me a reduced test case? I can't seem to reproduce that. What environment is this running in? Sounds browserless (?)

**Comment by hadimostafapour** [comment]
@jackdoyle yes, it's running in `NodeJS` environment.

As I said it was a server side rendering error, so there is no `window` object.
Passing ` _doc = window.document` as globals in UMD is the issue, that try to accessing window.document on construction,
so whenever package got imported and window object isn't available it throws error.

**Comment by hadimostafapour** [comment]
Ok, I just try to contribute a fix but I think repository codes are pre-built.

> For fix server-side-rendering issues we need to control variables initialization on construction.
> In UMD an IIFE called when module got loaded and either module exports.
> We have to import file because of isomorphic, but methods should not be called in server-side-rendering. So its not matter there is no DOM or window object.


So to fix IIFE In "Ex. TweenLite.js"
```
"use strict";
window = window || {}; // if window object is not available || browserless
var _exports = {},
    _doc =  window.document...
```

Exports at very top of file ( `_gsScope` was undefined in browserless environment, while TweenMaxBase needs it to be an object.)
```
export const _gsScope = (typeof(window) !== "undefined") ? window : (typeof(module) !== "undefined" && module.exports && typeof(global) !== "undefined") ? global : (typeof this === 'object' ? this : {});
```


Exports at very bottom of file "Weird :(, `_gsScope` is undefined (not an object) in browserless environment "

```
const isGsScopeAvailable = typeof _gsScope === 'object';

const gs = isGsScopeAvailable ? _gsScope.com.greensock : false;
export {TweenLite as default};
export const SimpleTimeline = isGsScopeAvailable ? gs.core.SimpleTimeline : false;
export const Animation = isGsScopeAvailable ? gs.core.Animation : false;
export const {Ease, Linear, Power0, Power1, Power2, Power3, Power4, TweenPlugin} = isGsScopeAvailable ? _gsScope : {};
export const EventDispatcher = isGsScopeAvailable ? gs.events.EventDispatcher : false;
```

**Comment by jackdoyle** [comment]
So you're saying that window, global, and "this" are all undefined in your environment? I'm not sure GSAP should be expected to work there, but might it be enough to just add " || {}" at the very end of the current _gsScope declaration? 

It would be super helpful if you could provide a reduced test case that'd allow me to see the error being thrown.

**Comment by jonatasnardi** [comment]
I'm getting same error, but I couldn't fix it. I'm building an Angular Universal server with NodeJS and webpack, running this command: _npm run build:client-and-server-bundles && npm run webpack:server_ 

This is the error message:
**_doc = window.document,
			    ^

TypeError: Cannot read property 'document' of undefined**

**Comment by jackdoyle** [comment]
@jonatasnardi any chance you (or anyone) could provide a reduced test case so that I could reproduce the error?

And you want to use GSAP in a completely browserless environment, right? For animating raw data?

**Comment by vasturiano** [comment]
@jackdoyle I'm having a similar issue in bundling a module that depends on TweenLite (built using Rollup), because of https://github.com/rollup/rollup/wiki/Troubleshooting#this-is-undefined

The error that occurs is:

> (!) `this` has been rewritten to `undefined`
https://github.com/rollup/rollup/wiki/Troubleshooting#this-is-undefined
node_modules/gsap/esm/TweenLite.js
19:   - remove the "export to multiple environments" in Definition().
20:  */
21: export const _gsScope = (typeof(window) !== "undefined") ? window : (typeof(module) !== "undefined" && module.exports && typeof(global) !== "undefined") ? global : this;

It leads to _gsScope being `undefined`. It seems that your suggestion above would fix the issue.

**Comment by jackdoyle** [comment]
This should be resolved in 2.0.1 (just published). Please let me know if you run into any other issues.
